### PR TITLE
Support using the login keychain

### PR DESCRIPTION
**Pull request by lox**
This adds `--use-login-keychain` which will use your default `login` keychain on macOS rather than creating an `aws-vault` keychain.

### CODE REVIEW

**Comment by mtibben** [comment]
I think I'm missing some context. What's the end goal, are we evaluating whether to use the login keychain as the default? And if so what's the migration path, any security tradeoffs?

**Comment by lox** [comment]
I guess let's talk about the opposite, what are the goals of a separate keychain? As I've spent more time with aws-vault, it just seems like a reasonable trade-off for UX to allow people to store (and eventually synchronise to iCloud) credentials in their primary login keychain. I see most of the benefit in aws-vault being the prompt for credential use. The seperate keychain offers much shorter timeouts, which in theory protects you if you leave you computer unlocked and unattended for more than 5 minutes, but it seems like a pretty marginal benefit. 

In the future, I'd like aws-vault to use the `login` keychain as the default as this is what most apps do and it's expected behaviour for most people. I'd rather push people to rotate credentials and use MFA liberally.

**Comment by lox** [comment]
Interested in @sj26 and @pda's take on this too.

File: cli/global.go [hunk 1]
```diff
@@ -42,17 +43,24 @@ func ConfigureGlobals(app *kingpin.Application) {
...
 		EnumVar(&GlobalFlags.PromptDriver, promptsAvailable...)
 
+	app.Flag("use-login-keychain", "Uses your default login keychain rather a new one").
+		OverrideDefaultFromEnvar("AWS_VAULT_USE_LOGIN_KEYCHAIN").
+		BoolVar(&GlobalFlags.UseLoginKeychain)
+
 	app.PreAction(func(c *kingpin.ParseContext) (err error) {
+		var keyringName = DefaultKeyringName
+		if GlobalFlags.UseLoginKeychain && GlobalFlags.Backend == keyring.KeychainBackend {
+			keyringName = "login"
```
Comment (mtibben) [inline cli/global.go]: Is `login` what we want here? Seems like it's possible for the "default" keychain to be named differently

**Comment by mtibben** [comment]
So to be clear the ACLs only allowing access to `aws-vault` are still set on the individual item right? This doesn't mean any old process can access the credentials

**Comment by lox** [comment]
Yup, correct. It's just which keychain it's located in.

File: cli/global.go (hunk 1 above)
Comment (lox) [reply cli/global.go]: Yeah, fair point. I will investigate that. Perhaps "default" keychain is the phrasing we should be using.

**Comment by mtibben** [comment]
That makes sense to me. Although I have found it useful in the past to back up / delete the aws-vault keychain directly when debugging aws-vault. I think I would want to retain the ability to specify the keychain in an env var as you had in #143 😬   Or do you have any other suggestions for that use case? I'm not convinced having `AWS_VAULT_KEYRING_NAME ` adds much complexity. Thoughts @pda ?

**Comment by lox** [comment]
Part of this is removing some of the complexity too. I'd like it to just use the default keychain eventually which should eliminate https://github.com/99designs/aws-vault/issues/49

**Comment by lox** [comment]
Perhaps an export or import facility would be better for that @mtibben?

**Comment by lox** [comment]
Could read/write to `~/.aws/credentials`.

**Comment by lox** [comment]
Deleting entries in the default keychain is still pretty easy.

File: cli/global.go (hunk 1 above)
Comment (sj26) [reply cli/global.go]: Looks a bit like you can use 'NULL' to mean 'default keychain':

https://developer.apple.com/documentation/security/1393225-seckeychainitemcreatefromcontent#topic-content

**Comment by sj26** [comment]
I'm on-board with default keychain :+1:

I kinda already have a patched version of aws-vault which changes KeychainName to login >.>

**Comment by heph** [comment]
I would love to see this as the default behavior to make the typical user experience as straightforward as possible, while keeping a flag or env var to override for people who require the flexibility.

I'm supporting 100+ devs who need to securely manage their AWS credentials with a minimum of fuss. Keeping credentials in a non-login keychain has and will result in them forgetting the password when "Always Allow" expires (as we recently found 356 days after first onboarding users with gpg keys). The login keychain is a perfect default as it's unlockable with the user's account password so cannot be forgotten.

**Comment by lox** [comment]
That makes a lot of sense @heph. We'll try and get this in the next major version. Thanks for the feedback!

**Comment by lox** [comment]
I'll progress this after #181, otherwise I don't have a good way of moving creds between keychains.

**Comment by lox** [comment]
Ended up deciding the import/export was a bad idea. Moving this to #197.

**Pull request closed without merging**